daisy/workflow/step.go

```diff
@@ -0,0 +1,154 @@
+package workflow
+
+import (
+	"errors"
+	"fmt"
+	"reflect"
+	"strings"
+	"time"
+)
+
+type stepImpl interface {
+	run(s *Step) error
+	validate(s *Step) error
+}
+
+// Step is a single daisy workflow step.
+type Step struct {
+	name string
+	w    *Workflow
```

Comment:
No, the actual *Step will be used. Our ctep types, or stepImpls, only use *Workflows currently. But I am making changes to validate methods that will register *Steps as a resource creator, deleter, or user. Without passing *Step, the validate methods have no idea what *Step they belong to.